modules/matchRoutes.js

```diff
@@ -2,18 +2,30 @@ import warning from './routerWarning'
 import { loopAsync } from './AsyncUtils'
 import { matchPattern } from './PatternUtils'
 import { createRoutes } from './RouteUtils'
+import { canUseMembrane } from './deprecateObjectProperties'
+import deprecateLocationProperties from './deprecateLocationProperties' 
 
-function getChildRoutes(route, location, callback) {
+function getChildRoutes(route, location, paramNames, paramValues, remainingPathname, callback) {
   if (route.childRoutes) {
     return [ null, route.childRoutes ]
   }
   if (!route.getChildRoutes) {
     return []
   }
 
-  let sync = true, result
+  let sync = true, result, progressStateWithLocation
+  const progressState = { 
```

Comment:
I would call this `partialNextState` – we use the term elsewhere for partial next state.

Can we omit `remainingPathname` here? I don't think it makes sense to pass down at this point. It feels odd for child routes to explicitly depend on the not-yet-matched portion of a route.